client/ayon_shotgrid/plugins/publish/collect_shotgrid_session.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+import ayon_api
+from ayon_shotgrid.lib.credentials import get_local_login
+
+import pyblish.api
+import shotgun_api3
+from shotgun_api3.shotgun import AuthenticationFault
+
+
+class CollectShotgridSession(pyblish.api.ContextPlugin):
+    """Collect shotgrid session using user credentials"""
+
+    order = pyblish.api.CollectorOrder
+    label = "Setting up Shotgrid user session."
+
+    def process(self, context):
+        self.log.info("Gathering The Shotgrid Session")
+        certificate_path = os.getenv("SHOTGUN_API_CACERTS")
+
+        if certificate_path is None or not os.path.exists(certificate_path):
```

Comment:
Set of `"SHOTGUN_API_CACERTS"` looks like something that should happen before a DCC is launched and definetelly should not happen in collector plugins.

Reply:
This was legacy, but can move it to the `lib/credentials`